### ISSUE TITLE
Crash when new commits were pushed while running CI (before running Danger)

### ISSUE THREAD

**Issue by maxfriedrich** [danger/danger#712]
# Report

## What did you do?

We're running Danger on Jenkins as a post-build action. Builds are started by the GitHub pull request builder plugin when we type phrases like "please test X" into GitHub Enterprise comments. I started a build with a comment and then, while the job was not yet done (and Danger had not yet run), I pushed a new commit.

## What did you expect to happen?

I expected Danger to update the commit status on the commit that was built in the Jenkins job.

## What happened instead?

Danger crashed:

```
+ bundle exec danger --new-comment
fatal: --unshallow on a complete repository does not make sense
bundler: failed to load command: danger (/Users/jenkins/.rvm/gems/ruby-2.3.1/bin/danger)
RuntimeError: Commit b6feb139 doesn't exist. Are you running `danger local/pr` against the correct repository? Also this usually happens when you rebase/reset and force-pushed.
  /Users/jenkins/.rvm/gems/ruby-2.3.1/gems/danger-4.0.5/lib/danger/scm_source/git_repo.rb:58:in `raise_if_we_cannot_find_the_commit'
  /Users/jenkins/.rvm/gems/ruby-2.3.1/gems/danger-4.0.5/lib/danger/scm_source/git_repo.rb:43:in `ensure_commitish_exists!'
  /Users/jenkins/.rvm/gems/ruby-2.3.1/gems/danger-4.0.5/lib/danger/request_sources/github.rb:77:in `setup_danger_branches'
  /Users/jenkins/.rvm/gems/ruby-2.3.1/gems/danger-4.0.5/lib/danger/danger_core/environment_manager.rb:57:in `ensure_danger_branches_are_setup'
  /Users/jenkins/.rvm/gems/ruby-2.3.1/gems/danger-4.0.5/lib/danger/danger_core/dangerfile.rb:258:in `setup_for_running'
  /Users/jenkins/.rvm/gems/ruby-2.3.1/gems/danger-4.0.5/lib/danger/danger_core/dangerfile.rb:269:in `run'
  /Users/jenkins/.rvm/gems/ruby-2.3.1/gems/danger-4.0.5/lib/danger/danger_core/executor.rb:27:in `run'
  /Users/jenkins/.rvm/gems/ruby-2.3.1/gems/danger-4.0.5/lib/danger/commands/runner.rb:66:in `run'
  /Users/jenkins/.rvm/gems/ruby-2.3.1/gems/claide-1.0.1/lib/claide/command.rb:334:in `run'
  /Users/jenkins/.rvm/gems/ruby-2.3.1/gems/danger-4.0.5/bin/danger:5:in `<top (required)>'
  /Users/jenkins/.rvm/gems/ruby-2.3.1/bin/danger:23:in `load'
  /Users/jenkins/.rvm/gems/ruby-2.3.1/bin/danger:23:in `<top (required)>'
```

`b6feb139` was the newly pushed commit that apparently doesn't exist yet in our checkout.

## Your Environment

- We're running Danger on Jenkins CI
- This happened with Danger 4.0.5 but the diff from 4.0.5 to 4.1.0 doesn't look like it would affect this.

**Comment by orta** [comment]
I wonder if this is the same as https://github.com/danger/danger/issues/660 - any chance you can try that?

**Comment by maxfriedrich** [comment]
I tried it with #662. Now there is no crash (:tada:), but Danger didn't run on the commit that the CI job was started with but on the newest commit. Jenkins sets the status to green on the first commit and Danger sets it to green on the newer commit. Can I specify which commit Danger should look at?

**Comment by orta** [comment]
Yep, you can pass in a base, and head to danger via the CLI ( check with `--help` for syntax I have the JS version installed globally ATM, so hard to check )

**Comment by maxfriedrich** [comment]
OK, thanks! I still do need #662 though, I tried it again with the current release and got this:
```
+ bundle exec danger --new-comment --head=0b71e64d207768becee06a6619c90e640bc5debd
fatal: --unshallow on a complete repository does not make sense
bundler: failed to load command: danger (/Users/jenkins/.rvm/gems/ruby-2.3.1/bin/danger)
RuntimeError: Commit 3a894fca doesn't exist. Are you running `danger local/pr` against the correct repository? Also this usually happens when you rebase/reset and force-pushed.
...
```
`3a894fca` is the newer commit again. Should we just `git pull` again before running Danger for now?

**Comment by orta** [comment]
That does sound like it, not really sure what a good answer is outside of that?

Depending on how set up you are in the Ruby version, you can use the JS version which cannot exhibit this problem ( it only uses the GH API to get this kind of metadata )

**Comment by maxfriedrich** [comment]
Fixed with #662 👍
